Sources/Common/Core/DataArray/index.js

```diff
@@ -136,6 +137,16 @@ function vtkDataArray(publicAPI, model) {
     publicAPI.modified();
   }
 
+  /**
+   * Reallocate model.values to a length of newSize and copy the old values to the new array.
+   * @param {Number} newSize
+   */
+  function reallocate(newSize) {
```

Comment:
Let's use the same reallocation scheme as in VTK C++, i.e. at least twice as much:
![image](https://user-images.githubusercontent.com/219044/181587564-f04dcc55-4c42-42e7-a324-7ab1c1b02e80.png)

Reply:
What about this?
```js
  /**
   * Reallocate model.values to a length of newSize and copy the old values to the new array.
   * @param {Number} requestedNumTuples
   */
  function resize(requestedNumTuples) {
    if (requestedNumTuples < 0) {
      // requestedNumTuples must be >= 0
      return false;
    }
    let numTuples = requestedNumTuples;
    const numComps = publicAPI.getNumberOfComponents();
    const curNumTuples = model.size / (numComps > 0 ? numComps : 1);

    if (numTuples > curNumTuples) {
      // Requested size is bigger than current size.  Allocate enough
      // memory to fit the requested size and be more than double the
      // currently allocated memory.
      numTuples += curNumTuples;
    } else if (numTuples === curNumTuples) {
      return true;
    } else {
      dataChange();
    }

    model.size = numTuples * numComps;
    const oldValues = model.values;
    model.values = macro.newTypedArray(model.dataType, model.size);
    model.values.set(oldValues);
    return true;
  }
```

Reply:
I changed it to this:

```js
  /**
   * Reallocate model.values and copy the old values to the new array.
   * @param {Number} requestedNumTuples
   */
  function reallocate(requestedNumTuples) {
    if (requestedNumTuples < 0) {
      // requestedNumTuples must be >= 0
      return false;
    }
    const numComps = publicAPI.getNumberOfComponents();
    const curNumTuples = model.values.length / (numComps > 0 ? numComps : 1);

    let newSize;
    if (requestedNumTuples > curNumTuples) {
      // Requested size is bigger than current size.  Allocate enough
      // memory to fit the requested size and be more than double the
      // currently allocated memory.
      newSize = (requestedNumTuples + curNumTuples) * numComps;
    } else {
      return true;
    }

    const oldValues = model.values;
    model.values = macro.newTypedArray(model.dataType, newSize);
    model.values.set(oldValues);
    return true;
  }
```

Reply:
```
if (requestedNumTuples <= curNumTuples) {
  return true;
}
// Requested size is bigger than current size.  Allocate enough
// memory to fit the requested size and be more than double the
// currently allocated memory.
const newSize = (requestedNumTuples + curNumTuples) * numComps;
```